### ISSUE TITLE
OBL_UNSATISFIED_OBLIGATION not tracked for Connection.prepareCall, only for Connection.prepareStatement

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4148]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Check name

`OBL_UNSATISFIED_OBLIGATION`

### Description

The obligation-tracking detector flags a method that obtains a `java.sql.Statement` from `Connection.prepareStatement(...)` and never closes it, but it does not flag the same code when the statement is obtained from `Connection.prepareCall(...)`. Both factory methods return a `java.sql.Statement` subtype (`PreparedStatement` and `CallableStatement` respectively) whose `close()` obligation is exactly what this detector is documented to track — and the warning text itself reads "may fail to clean up java.sql.Statement". The two forms are equivalent with respect to the tracked obligation: in both cases a `Statement` resource is created and leaked. SpotBugs' obligation policy database recognizes `prepareStatement` as creating an obligation but is missing `prepareCall`, so the leak is silently missed for callable statements.

### Reproducer

```java
// BEFORE — prepareStatement, 1 report
import java.sql.Connection;
import java.sql.Statement;
import java.sql.SQLException;

public class DeadStore {
    public void testDatabaseStore(Connection c) throws SQLException {
        Statement s = c.prepareStatement("select something from other");
    }
}
```

```java
// AFTER — prepareCall (also returns a Statement that is leaked), 0 reports
import java.sql.Connection;
import java.sql.Statement;
import java.sql.SQLException;

public class DeadStore {
    public void testDatabaseStore(Connection c) throws SQLException {
        Statement s = c.prepareCall("select something from other");
    }
}
```

### Expected behavior

`OBL_UNSATISFIED_OBLIGATION` should report the leaked `Statement` for both `prepareStatement` and `prepareCall`, since both create a `java.sql.Statement` subtype that carries the same unclosed-resource obligation.

### Actual behavior

- BEFORE (`prepareStatement`): 1 report — `DeadStore.testDatabaseStore(Connection) may fail to clean up java.sql.Statement`.
- AFTER (`prepareCall`): 0 reports.
